parser-typechecker/src/Unison/Codebase/FileCodebase/Common.hs

```diff
@@ -238,7 +234,8 @@ encodeFileName = let
                  = "$x" <> encodeHex [c] <> "$" <> go rem
                | otherwise = c : go rem
   go [] = []
-  encodeHex = Text.unpack . decodeUtf8 . Hex.hex . encodeUtf8 . Text.pack
+  encodeHex :: String -> String
+  encodeHex = Text.unpack . Text.encodeBase16 . Text.pack
```

Comment:
Within `FileCodebase`, and `Serialization`, we have to be careful even changing existing behavior at all, even to fix bugs, as it could cause some existing user data to become unreadable.  I think this change is safe though, because we don't have any builtin names that use extended characters, and because git caches can be missed safely.

Reply:
Good point. In this case, there actually was a change: `hex` uppercase letters `A-F`, but `base16` outputs lowercase letters `a-f`. Is that a problem?

Reply:
@aryairani I think whether the lowercase/uppercase switch is a problem is the only remaining issue. Tests pass, so I'm not sure :)

Reply:
Hm this morning I’m only “pretty sure” it’s okay too, so I think we should revert this to be safe, until the next codebase format.

Reply:
(In both encodeFileName and decodeFileName)

Reply:
@mitchellwrosen Sorry, I hadn't spotted your question until now. 

I'm not sure; let's throw in a `toUpper` to match the old behavior (to be safe), and then if we have an analogous function in the V2 codebase, we should use lower case letters, because it looks nicer IMO.

Reply:
Btw, good catch on the `A-F` vs `a-f`.

Reply:
`toUpper` added!